slackscot.go

```diff
@@ -852,16 +934,16 @@ func (s *Slackscot) routeMessage(me slack.MessageEvent) (responses []OutgoingMes
 	responses = make([]OutgoingMessage, 0)
 
 	// Ignore messages_replied and messages send by "us"
-	if m.User == s.selfID || m.BotID == s.selfID || m.BotID == s.selfBotID {
-		s.log.Debugf("Ignoring message from user [%s] / bot ID [%s] because that's \"us\" (userID: [%s], botID: [%s]", m.User, m.BotID, s.selfID, s.selfBotID)
+	if s.botMatcher.IsBot(m) {
+		s.log.Debugf("Ignoring message from user [%s] / bot ID [%s] because that's \"us\" (%s)", m.User, m.BotID, s.botMatcher)
 
 		return responses
 	}
 
 	// Try commands or hear actions depending on the format of the message
-	if isCommand, isDM := isCommand(m, s.selfUserPrefix); isCommand {
+	if s.isCommand(m) {
```

Comment:
It doesn't seem worth removing the clarity of the two functions to avoid running a trim twice. Thoughts?

Reply:
The reason I was suggesting removing the `isCommand` wrapping function is because I was picturing the `cmdMatcher` being now responsible for determining if a message was a command or not and that `func (s *Slackscot) isCommand` function doesn't really do much now. If you think that syntax is heavy, you could also just do and still remove the `isCommand`:
```
isCmd := s.cmdMatcher.IsCmd(m) || isDirectMessage(m)
if isCmd {
}
```

Reply:
I was for keeping it as it keeps the user of IsCmd from having to know about cmdMatcher directly. Less typing, less chance of someone using the method from messing something up. Or forgetting the isDirectMessage bit.